### ISSUE TITLE
Deploy Pod Security Policy with Argo

### ISSUE THREAD

**Issue by saschagrunert** [argoproj/argo#2906]
# Summary
Deploy Pod Security Policy with Argo to allow SYS_PRACE when using the PNS executor.
# Motivation

Argo needs `SYS_PTRACE` as capability when using the PNS executor, which may be blocked by default PSPs. To have a better out-of-the-box deployment experience we could define our own PSP and ship it with argo.

# Proposal

We add the PSP to the deployment link the cluster role(s) to it.

**Comment by Joibel** [comment]
Closing as obsolete.

This asked for a PodSecurityPolicy shipped with Argo so the PNS executor could get `SYS_PTRACE`. Neither side of that still exists:

- The PNS executor was removed in v3.4.0 (#8296, fixes #7804). Emissary is
the only executor now and does not require `SYS_PTRACE`.
- PodSecurityPolicy was deprecated in Kubernetes 1.21 and removed in 1.25,
so we cannot ship one for any currently supported cluster.

If you are hardening an Argo install today, the equivalent is running under the restricted Pod Security Standard via `securityContext` / `seccompProfile` on the controller and workflow pods. Please open a new issue if something there is missing.
